Automattic/wp-calypso
# Replce composite-checkout-demo with a Storybook

#### Background

`@automattic/composite-checkout` provides a demo of the component. This demo is based on a custom webpack config launched by `webpack-dev-server`. The migration of this package to `webpack 5` was really slow (in fact, it is still in `beta.2`), and given we only use it for the `composite-checkout` demo, I've been looking to replace it since the `webpack 5` migration.

#### Changes proposed in this Pull Request

This PR replaces the demo with a Storybook.

#### Testing instructions

Checkout this branch and launch the storybook with `yarn composite-checkout:storybook:start`. It should open Storybook with a `composite-checkout` component. You should be able to simulate a payment using a fake credit card.

![image](https://user-images.githubusercontent.com/975703/115686890-f1dad780-a359-11eb-87ec-6365ad8d59d6.png)

Compare the behaviour with the old demo (run `yarn composite-checkout-demo` in `trunk`).
